components/ide-service/pkg/server/server.go

```diff
@@ -445,27 +460,23 @@ func (s *IDEServiceServer) ResolveWorkspaceConfig(ctx context.Context, req *api.
 			}
 		}
 
-		// we always need WebImage for when the user chooses a desktop ide
-		resp.WebImage = getUserIDEImage(defaultIde)
-		resp.IdeImageLayers = getUserImageLayers(defaultIde)
-
 		var desktopImageLayer string
 		var desktopUserImageLayers []string
 		if chosenIDE.Type == config.IDETypeDesktop {
-			desktopImageLayer = getUserIDEImage(chosenIDE)
-			desktopUserImageLayers = getUserImageLayers(chosenIDE)
+			desktopImageLayer = getUserIDEImage(resultingIdeName)
```

Comment:
I remember we used to have golden tests for that? Could we add more tests to cover new cases?